packages/frontend/src/document/theory_selector.tsx

```diff
@@ -0,0 +1,76 @@
+import { For, createMemo } from "solid-js";
+import type { TheoryLibrary } from "../stdlib/types";
+import type { ModelDocument } from "./types";
+import "./model_document_editor.css";
+import type { TheoryMeta } from "../stdlib/types";
+
+interface TheorySelectorProps {
+    docHandle: any;
+    theories: TheoryLibrary;
+    doc: ModelDocument;
+}
+
+const TheorySelector = (props: TheorySelectorProps) => {
+    const groupedTheories = createMemo(() => {
+        const theories = Array.from(props.theories.metadata()).filter(
+            (meta) => meta.divisionCategory,
+        );
```

Comment:
Is this filtering needed? Seems like you're already checking for theories without a `divisionCategory` below.